### PR TITLE
RK-8248 - Fix permissions and expose inMemoryDB flag

### PR DESCRIPTION
**Pull request by adilrookout**

### CODE REVIEW

File: charts/controller/templates/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,8 @@ spec:
 {{ toYaml .Values.podAnnotations | indent 8 }}
 {{- end }}
     spec:
+      securityContext:
+        fsGroup: 1000
```
Comment (orishoshan) [inline charts/controller/templates/deployment.yaml:31]: What is this magic number? What does this do?

File: charts/datastore/templates/service.yaml [hunk 2]
```diff
@@ -17,12 +17,12 @@ spec:
     {{ if or (eq .Values.datastore.serverMode "PLAIN") (eq .Values.datastore.serverMode "AUTOTLS") }} # AUTOTLS mode requires port 80 for communicating with LetsEncrypt
     - protocol: TCP
       port: 80
-      targetPort: 80
+      targetPort: 8080
```
Comment (orishoshan) [inline charts/datastore/templates/service.yaml:20]: How could this possibly be the same port if there are multiple servers on the DOP?

File: charts/datastore/templates/persistentVolume.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{- if (eq .Values.datastore.inMemoryDb false) }}
```
Comment (orishoshan) [inline charts/datastore/templates/persistentVolume.yaml:1]: What if inMemoryDb is not defined? This condition means it will make an inMemoryDb.
I suggested using `eq` if you wanna make sure it's exactly false, but it needs to make sense to do that in the context of what you're trying to do

File: charts/datastore/templates/service.yaml (hunk 2 above)
Comment (adilrookout) [reply charts/datastore/templates/service.yaml:20]: WDYM multiple servers on the DOP ?
DOP is one pod with one container

Comment (orishoshan) [reply charts/datastore/templates/service.yaml:20]: It runs multiple servers in AUTOTLS mode. Take a closer look at this part you changed, it would have had both 80 and 443 at the same time - both here in the yaml and in the code. It can't be the same port.

File: charts/datastore/values.yaml [hunk 4]
```diff
@@ -4,8 +4,10 @@
 
 image:
   repository: rookout/data-on-prem
+  pullPolicy: Always
```
Comment (Gilwe) [inline charts/datastore/values.yaml:7]: Add this to the readme

Comment (adilrookout) [reply charts/datastore/values.yaml:7]: Why ? it's not configurable

File: charts/datastore/templates/persistentVolume.yaml (hunk 3 above)
Comment (adilrookout) [reply charts/datastore/templates/persistentVolume.yaml:1]: it always defined in the `values.yaml`

File: charts/datastore/templates/deployment.yaml [hunk 5]
```diff
@@ -49,28 +55,36 @@ spec:
...
+              value: "True"
+{{- end }}
 {{- if (.Values.datastore.url_prefix) }}
             - name: ROOKOUT_DOP_URL_PREFIX
               value: {{ .Values.datastore.url_prefix | quote }}
 {{- end }}
             - name: ROOKOUT_DOP_SERVER_MODE
               value: {{ .Values.datastore.serverMode | quote }}
+            - name: ROOKOUT_DOP_PORT
+              value: .Values.datastore.dopContainerPort
```
Comment (Gilwe) [inline charts/datastore/templates/deployment.yaml:69]: I think this is a mistake, the ROOKOUT_DOP_PORT determines the outside port, not the inner one

File: charts/datastore/README.md [hunk 6]
```diff
@@ -43,6 +43,11 @@ The data-on-prem solution runs with one of 3 modes (datastore.serverMode):
 | datastore.token | Your organization token. This is the same token as that used by your Rookout ETL controller |
 | datastore.tokenFromSecret.name| Secret ref in which the Rookout token resides |
 | datastore.tokenFromSecret.key| Key of the secret in which the Rookout token resides |
+| datastore.dopContainerPort| Main container port, default: 8080 |
+| datastore.dopServicePort| Main service port, default: 8080 |
```
Comment (Gilwe) [inline charts/datastore/README.md:47]: what is this? this isnt used

File: charts/datastore/templates/service.yaml [hunk 7]
```diff
@@ -14,15 +14,21 @@ spec:
...
-      targetPort: 80
-      name: {{ printf "%s-http" (include "datastore.name" .) }}
+      targetPort: {{ .Values.datastore.letsEncryptContainerlPort }}
+      name: {{ printf "%s-lets-encrypt" (include "datastore.name" .) }}
     {{ end }}
-    {{ if or (eq .Values.datastore.serverMode "TLS") (eq .Values.datastore.serverMode "AUTOTLS") }}
+    {{- if or (eq .Values.datastore.serverMode "TLS") (eq .Values.datastore.serverMode "AUTOTLS") }}
     - protocol: TCP
       port: 443
-      targetPort: 443
```
Comment (Gilwe) [inline charts/datastore/templates/service.yaml:26]: why is this removed?

Comment (adilrookout) [reply charts/datastore/templates/service.yaml:26]: it's not removed it's defined at the bottom in line 32

Comment (adilrookout) [reply charts/datastore/templates/service.yaml:26]: the `targetPort` of the dop is the same on both modes, just the service port changes and not configurable

File: charts/datastore/templates/persistentVolume.yaml (hunk 3 above)
Comment (orishoshan) [reply charts/datastore/templates/persistentVolume.yaml:1]: What if they don't use the values.yaml? We do this regularly when we don't use it as a chart but as yamls...

File: charts/datastore/README.md [hunk 8]
```diff
@@ -43,6 +43,9 @@ The data-on-prem solution runs with one of 3 modes (datastore.serverMode):
 | datastore.token | Your organization token. This is the same token as that used by your Rookout ETL controller |
 | datastore.tokenFromSecret.name| Secret ref in which the Rookout token resides |
 | datastore.tokenFromSecret.key| Key of the secret in which the Rookout token resides |
+| datastore.dopContainerPort| Main container port, default: 8080 |
```
Comment (Gilwe) [inline charts/datastore/README.md:46]: it is used by both the external and the internal port, what's the reason for that? 
right now the nonroot testing open port 80 and 443 normally

File: charts/datastore/templates/persistentVolume.yaml (hunk 3 above)
Comment (adilrookout) [reply charts/datastore/templates/persistentVolume.yaml:1]: but here we are talking on a chart. 

In yaml you can do whatever you want and i have no way to save you with render-time checks opposed to chart.

Comment (orishoshan) [reply charts/datastore/templates/persistentVolume.yaml:1]: I don't understand, why not use `not`?? It would cover the case of having it not defined and behave correctly...

File: charts/datastore/README.md (hunk 8 above)
Comment (adilrookout) [reply charts/datastore/README.md:46]: this variable only affects the internal port

Comment (adilrookout) [reply charts/datastore/README.md:46]: external port not controlled by any variable but hardcoded and determined by the server mode

File: charts/datastore/templates/persistentVolume.yaml (hunk 3 above)
Comment (adilrookout) [reply charts/datastore/templates/persistentVolume.yaml]: used default instead

File: charts/datastore/README.md (hunk 8 above)
Comment (Gilwe) [reply charts/datastore/README.md:46]: it seems not:
Screen Shot 2021-01-13 at 9 28 57

Comment (Gilwe) [reply charts/datastore/README.md:46]: https://github.com/Rookout/helm-charts/pull/19#discussion_r555948825

File: charts/datastore/templates/deployment.yaml [hunk 9]
```diff
@@ -26,13 +26,19 @@ spec:
 {{ toYaml .Values.podAnnotations | indent 8 }}
 {{- end }}
     spec:
+      securityContext:
+        # group ID 1000 is associated with all Containers in the Pod automatically by k8s
+        # we use it to give non-root user access to mounted volumes
+        fsGroup: 1000
```
Comment (orishoshan) [inline charts/datastore/templates/deployment.yaml:32]: fsGroup should be the same id as the group id you set in the Docker image, and the same group id Kubernetes runs it as (use runAsGroup)

**Review by orishoshan — APPROVED** [review]

**Review by Gilwe — APPROVED** [review]

**Review by Gilwe — APPROVED** [review]
